pkg/kp/pcstore/consul_store.go

```diff
@@ -44,9 +45,24 @@ func (s *consulStore) Create(
 	clusterName fields.ClusterName,
 	podSelector klabels.Selector,
 	annotations fields.Annotations,
+	session Session,
 ) (fields.PodCluster, error) {
 	id := fields.ID(uuid.New())
 
+	unlocker, err := s.lockForUpdateCreation(podID, availabilityZone, clusterName, session)
```

Comment:
I just realized this was named this way after "rolling updates" not "update or creation"